subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcPluginVersionIntegrationTest.groovy

```diff
@@ -48,7 +51,7 @@ class CodeNarcPluginVersionIntegrationTest extends MultiVersionIntegrationSpec {
             ${JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_14) ?
             """
             configurations.codenarc {
-                resolutionStrategy.force 'org.codehaus.groovy:groovy:${GroovySystem.version}'
+                resolutionStrategy.force 'org.codehaus.groovy:groovy:3.0.11' // force latest Groovy 3 when using Java 14+.  Do not use GroovySystem#version as Groovy 4 needs different coordinates
```

Comment:
Is there a reason you didn't use `3.0.12` here? Or even better, `GroovyCoverage.MINIMAL_GROOVY_3`?

Reply:
Good idea. At the time, 3.0.11 was the newest.